src/runtime-rs/crates/shimv2/src/bin/shimv2.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    ffi::{OsStr, OsString},
+    path::PathBuf,
+};
+
+use anyhow::{anyhow, Context, Result};
+use nix::{
+    mount::{mount, MsFlags},
+    sched::{self, CloneFlags},
+};
+use shimv2::{Args, Error, ShimExecutor};
+
+const DEFAULT_RUNTIME_WORKER_THREADS: usize = 2;
+const ENV_RUNTIME_WORKER_THREADS: &str = "RUNTIME_WORKER_THREADS";
+
+#[derive(Debug)]
+enum Action {
+    Run(Args),
+    Start(Args),
+    Delete(Args),
+    Help,
+    Version,
+    ShimVersion,
+}
+
+fn parse_args(args: &[OsString]) -> Result<Action> {
+    let mut help = false;
+    let mut version = false;
+    let mut shim_version = false;
+    let mut shim_args = Args::default();
+
+    // Crate `go_flag` is used to keep compatible with go/flag package.
+    let rest_args = go_flag::parse_args_with_warnings::<String, _, _>(&args[1..], None, |flags| {
+        flags.add_flag("address", &mut shim_args.address);
+        flags.add_flag("bundle", &mut shim_args.bundle);
+        flags.add_flag("debug", &mut shim_args.debug);
+        flags.add_flag("id", &mut shim_args.id);
+        flags.add_flag("namespace", &mut shim_args.namespace);
+        flags.add_flag("publish-binary", &mut shim_args.publish_binary);
+        flags.add_flag("socket", &mut shim_args.socket);
+        flags.add_flag("help", &mut help);
+        flags.add_flag("version", &mut version);
+        flags.add_flag("v", &mut shim_version);
+    })
+    .context(Error::ParseArgument(format!("{:?}", args)))?;
+
+    if help {
+        Ok(Action::Help)
+    } else if version {
+        Ok(Action::Version)
+    } else if shim_version {
+        Ok(Action::ShimVersion)
+    } else if rest_args.is_empty() {
+        Ok(Action::Run(shim_args))
+    } else if rest_args[0] == "start" {
+        Ok(Action::Start(shim_args))
+    } else if rest_args[0] == "delete" {
+        Ok(Action::Delete(shim_args))
+    } else {
+        Err(anyhow!(Error::InvalidArgument))
+    }
+}
+
+fn show_help(cmd: &OsStr) {
+    let path = PathBuf::from(cmd);
+    let name = match path.file_name() {
+        Some(v) => v.to_str(),
+        None => None,
+    };
+
+    let name = name.unwrap_or("containerd-shim-kata-v2");
+
+    eprintln!(
+        r#"Usage of {}:
+  -address string
+        grpc address back to main containerd
+  -bundle string
+        path to the bundle if not workdir
+  -debug
+        enable debug output in logs
+  -id string
+        id of the task
+  -namespace string
+        namespace that owns the shim
+  -publish-binary string
+        path to publish binary (used for publishing events) (default "containerd")
+  -socket string
+        socket path to serve
+  --version
+        show the shim version detail and exit
+  -v    show the shim version and exit    
+"#,
+        name
+    );
+}
+
+const DEFAULT_PROJECT: &str = "Kata Containers";
```

Comment:
Do we want to call out that its `runtime-rs` here or will that be covered elsewhere?

Reply:
I think Kata Container is the parent project, runtime-rs is a part of Kata Containers